ddl/ddl_worker.go

```diff
@@ -369,32 +369,41 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
 		startTS = txn.StartTS()
 		return nil
 	})
-	if err == nil {
-		jobTasks := make([]*model.Job, 0, len(tasks))
-		for i, task := range tasks {
-			job := task.job
-			job.Version = currentVersion
-			job.StartTS = startTS
-			job.ID = ids[i]
-			setJobStateToQueueing(job)
+	if err != nil {
+		return errors.Trace(err)
+	}
 
-			if d.stateSyncer.IsUpgradingState() && !hasSysDB(job) {
+	jobTasks := make([]*model.Job, 0, len(tasks))
+	for i, task := range tasks {
+		job := task.job
+		job.Version = currentVersion
+		job.StartTS = startTS
+		job.ID = ids[i]
+		setJobStateToQueueing(job)
+
+		if d.stateSyncer.IsUpgradingState() {
+			hasSysDB, err := util.HasSysDB(job)
```

Comment:
Another is check the `HasSysDB`'s error.

Reply:
Maybe wrap this block into a separate function `trySetJobToPausing()` or `setJobPausingForUpgrade()`?